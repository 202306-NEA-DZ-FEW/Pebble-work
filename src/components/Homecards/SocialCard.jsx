import React from "react";

const SocialCardData = [
    {
        name: "Emilia W.",
        occupation: "Volunteer",
        text: "Thanks to PebbleWork I'm able to make a positive impact on my community and the people around me. I'm thankful to have a website where we can all come together and TRULY change things.",
        imageSrc: "./images/icon_default.png",
    },
    {
        name: "Mubashir Aazar",
        occupation: "Beyrouth Youth Club rep.",
        text: " I must commend Pebble Work for being a game-changer in our mission for positive community impact. Kudos to Pebble Work for providing us with the tools to transform our vision into meaningful actions.",
        imageSrc: "./images/Aazar_BYC.png",
    },
    {
        name: "Ali",
        occupation: "occupation",
        text: "This is the third testimonial. Yet another testimonial here.",
        imageSrc: "./images/icon_default.png",
    },
];

const SocialCardSection = () => {
    return (
        <div className=' w-full max-w-[1440px] px-5 md:px-[123px] lg:px-[127px] bg-white bg-opacity-50 flex flex-wrap justify-center items-center gap-10 py-8 rounded-md'>
            <div className='w-full text-center text-[#2E72AA] text-4xl font-medium leading-[44px] tracking-tight'>
                Hear it from our volunteers
            </div>
            {SocialCardData.map((SocialCard, index) => (
                <div
                    key={index}
                    className='bg-white p-3 rounded-xl w-full md:w-1/3 lg:w-1/4 flex flex-col items-center gap-5'
                >
                    <img
                        loading='lazy'
                        className='w-20 h-20 rounded-full'
                        src={SocialCard.imageSrc}
                        alt={`Image ${index}`}
                    />
                    <div className='w-full text-center'>
                        <div className='text-zinc-900 text-xl md:text-lg lg:text-xl font-medium leading-[30px] tracking-tight'>
                            {SocialCard.name}
                            <br />
                            <span className='italic font-normal text-[#2E72AA]'>
                                {SocialCard.occupation}
                            </span>
                        </div>
                        <div className='text-zinc-500 text-base md:text-md lg:text-lg font-normal leading-[30px] tracking-tight'>
                            {SocialCard.text}
                        </div>
                    </div>
                </div>
            ))}
        </div>
    );
};

export default SocialCardSection;
