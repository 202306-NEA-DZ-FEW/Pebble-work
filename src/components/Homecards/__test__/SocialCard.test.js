import React from "react";
import { render } from "@testing-library/react";
import SocialCard from "@/components/Homecards/SocialCard";

describe("SocialCard", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<SocialCard />);
        expect(asFragment()).toMatchSnapshot();
    });
});
