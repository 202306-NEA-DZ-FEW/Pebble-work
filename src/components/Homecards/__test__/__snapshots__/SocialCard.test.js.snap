// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="shadow-md w-full max-w-[1440px] px-5 md:px-[123px] lg:px-[127px] bg-white flex flex-wrap justify-center items-center gap-10"
  >
    <div
      class="w-full text-center text-zinc-900 text-4xl font-medium leading-[44px] tracking-tight"
    >
      Hear it from volunteers
    </div>
    <div
      class="w-full md:w-1/3 lg:w-1/4 flex flex-col items-center gap-5"
    >
      <img
        alt="Image 0"
        class="w-16 h-16 rounded-full"
        loading="lazy"
        src="https://s3-alpha-sig.figma.com/img/c5db/07f5/e831b1a64ef23d6ce6baa81fbcf33e56?Expires=1698624000&Signature=ME0PjPYRWZJDhMo4ykxlAwRO~AxZpiHQvbSgONHGvYuj4zg7HnxgNi9oXEVcCWHZpmYteXjJ2jrPNdleu~MrphbU3MWGNCf2U~DR1rMOE7WIoFMBROkUj4P6CX9ffcSZoSTvdOUL2bI17RAVEknuPIIHeJeuC4YNptBmhmPVqIFR9A2UjHB~btkjTqPkxWuYZCjXo0Z1hzlCfbuCRN7REtvsS2k3N52u0b6Ku4ODbAuNvjaq7Ra1q7IiMPcnigMlxoZ3NVgVZ2hyfJKRyTVibPptohLY-68dLWgaufNU4U6Hx8zr9e~hKFNBxhfE2QG4hrJ5w6SC0KiXuX3uIbSIxg__&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4"
      />
      <div
        class="w-full text-center"
      >
        <div
          class="text-zinc-900 text-xl md:text-2xl lg:text-3xl font-medium leading-[30px] tracking-tight"
        >
          Quby, occupation
        </div>
        <div
          class="text-zinc-500 text-base md:text-xl lg:text-2xl font-normal leading-[30px] tracking-tight"
        >
          This is the first testimonial. It can be about anything.
        </div>
      </div>
    </div>
    <div
      class="w-full md:w-1/3 lg:w-1/4 flex flex-col items-center gap-5"
    >
      <img
        alt="Image 1"
        class="w-16 h-16 rounded-full"
        loading="lazy"
        src="https://s3-alpha-sig.figma.com/img/c5db/07f5/e831b1a64ef23d6ce6baa81fbcf33e56?Expires=1698624000&Signature=ME0PjPYRWZJDhMo4ykxlAwRO~AxZpiHQvbSgONHGvYuj4zg7HnxgNi9oXEVcCWHZpmYteXjJ2jrPNdleu~MrphbU3MWGNCf2U~DR1rMOE7WIoFMBROkUj4P6CX9ffcSZoSTvdOUL2bI17RAVEknuPIIHeJeuC4YNptBmhmPVqIFR9A2UjHB~btkjTqPkxWuYZCjXo0Z1hzlCfbuCRN7REtvsS2k3N52u0b6Ku4ODbAuNvjaq7Ra1q7IiMPcnigMlxoZ3NVgVZ2hyfJKRyTVibPptohLY-68dLWgaufNU4U6Hx8zr9e~hKFNBxhfE2QG4hrJ5w6SC0KiXuX3uIbSIxg__&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4"
      />
      <div
        class="w-full text-center"
      >
        <div
          class="text-zinc-900 text-xl md:text-2xl lg:text-3xl font-medium leading-[30px] tracking-tight"
        >
          Taki, occupation
        </div>
        <div
          class="text-zinc-500 text-base md:text-xl lg:text-2xl font-normal leading-[30px] tracking-tight"
        >
          This is the second testimonial. It can be about anything else.
        </div>
      </div>
    </div>
    <div
      class="w-full md:w-1/3 lg:w-1/4 flex flex-col items-center gap-5"
    >
      <img
        alt="Image 2"
        class="w-16 h-16 rounded-full"
        loading="lazy"
        src="https://s3-alpha-sig.figma.com/img/c5db/07f5/e831b1a64ef23d6ce6baa81fbcf33e56?Expires=1698624000&Signature=ME0PjPYRWZJDhMo4ykxlAwRO~AxZpiHQvbSgONHGvYuj4zg7HnxgNi9oXEVcCWHZpmYteXjJ2jrPNdleu~MrphbU3MWGNCf2U~DR1rMOE7WIoFMBROkUj4P6CX9ffcSZoSTvdOUL2bI17RAVEknuPIIHeJeuC4YNptBmhmPVqIFR9A2UjHB~btkjTqPkxWuYZCjXo0Z1hzlCfbuCRN7REtvsS2k3N52u0b6Ku4ODbAuNvjaq7Ra1q7IiMPcnigMlxoZ3NVgVZ2hyfJKRyTVibPptohLY-68dLWgaufNU4U6Hx8zr9e~hKFNBxhfE2QG4hrJ5w6SC0KiXuX3uIbSIxg__&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4"
      />
      <div
        class="w-full text-center"
      >
        <div
          class="text-zinc-900 text-xl md:text-2xl lg:text-3xl font-medium leading-[30px] tracking-tight"
        >
          Ali, occupation
        </div>
        <div
          class="text-zinc-500 text-base md:text-xl lg:text-2xl font-normal leading-[30px] tracking-tight"
        >
          This is the third testimonial. Yet another testimonial here.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
